Blockzilla/IntroViewController.swift

```diff
@@ -26,9 +26,88 @@ struct IntroViewControllerUX {
     static let FadeDuration = 0.25
 }
 
+protocol PageControlDelegate {
+    func incrementPage()
+    func decrementPage()
+}
+
+class PageControl: UIView {
+    
+    var currentPage = 0 {
+        didSet {
+            selectIndex(currentPage)
+        }
+    }
+    
+    var numberOfPages = 0 {
+        didSet {
+            addPages()
+        }
+    }
+    
+    var stack: UIStackView?
+    var delegate: PageControlDelegate?
+    
+    func addPages() {
+        var buttonArray = [UIButton]()
+        
+        // Ensure we have at least one button
+        if numberOfPages == 0 {
+            return
+        }
+        
+        for _ in 0..<numberOfPages {
+            let button = UIButton(frame: CGRect(x: 0, y: 0, width: 6, height: 6))
+            button.setImage(UIImage(imageLiteralResourceName: "page_indicator"), for: .normal)
+            buttonArray.append(button)
+        }
+        
+        // Enable the buttons to be tapped to switch to a new page
+        buttonArray.forEach({ button in
+            button.addTarget(self, action: #selector(selected(sender:)), for: .touchUpInside)
+        })
+        
+        stack = UIStackView(arrangedSubviews: buttonArray)
+        stack?.spacing = 20
+        stack?.distribution = .equalCentering
+        stack?.alignment = .center
+        
+        currentPage = 0
+    }
+    
+    func selectIndex(_ index:Int){
+        let buttons = stack?.arrangedSubviews as! [UIButton]
+        for button in buttons {
+            button.isSelected = false
+            button.alpha = 0.3
+        }
+        
+        buttons[index].isSelected = true
+        buttons[index].alpha = 1
+    }
+    
+    @objc func selected(sender : UIButton){
+        for i in 0..<((stack?.arrangedSubviews)! as! [UIButton]).count {
```

Comment:
avoid force-unwrap and force-cast, `guard let count = ... else { return }` or `else { assert(false); return }`